src/overlay_12_0224E4FC.c

```diff
@@ -0,0 +1,1224 @@
+        }
+    }
+    
+    if (heldItem2 == HOLD_EFFECT_SPEED_DOWN) {
+        loweredPriority2 = 1;
+    }
+    
+    ctx->unk_21F0[battlerId1] = speed1;
+    ctx->unk_21F0[battlerId2] = speed2;
+    
+    if (!flag) {
+        action1 = ctx->unk_21A8[battlerId1][3];
+        action2 = ctx->unk_21A8[battlerId2][3];
+        movePos1 = ctx->movePos[battlerId1];
+        movePos2 = ctx->movePos[battlerId2];
+        if (action1 == 1) { //fight button
+            if (ctx->turnData[battlerId1].struggleFlag) {
+                moveNo1 = MOVE_STRUGGLE;
+            } else {
+                moveNo1 = GetBattlerVar(ctx, battlerId1, BMON_DATA_MOVE1 + movePos1, NULL);
+            }
+        }
+        if (action2 == 1) { //fight button
+            if (ctx->turnData[battlerId2].struggleFlag) {
+                moveNo2 = MOVE_STRUGGLE;
+            } else {
+                moveNo2 = GetBattlerVar(ctx, battlerId2, BMON_DATA_MOVE1 + movePos2, NULL);
+            }
+        }
+        movePriority1 = ctx->unk_334.moveData[moveNo1].priority;
+        movePriority2 = ctx->unk_334.moveData[moveNo2].priority;
+    }
+    
+    if (movePriority1 == movePriority2) {
+        if (boostedPriority1 && boostedPriority2) {
+            if (speed1 < speed2) {
+                ret = 1;
+            } else if (speed1 == speed2 && BattleSys_Random(bsys)&1) {
+                ret = 2;
+            }
+        } else if (!boostedPriority1 && boostedPriority2) {
+            ret = 1;
+        } else if (boostedPriority1 && !boostedPriority2) {
+            ret = 0;
+        } else if (loweredPriority1 && loweredPriority2) {
+            if (speed1 > speed2) {
+                ret = 1;
+            } else if (speed1 == speed2 && BattleSys_Random(bsys)&1) {
+                ret = 2;
+            }
+        } else if (loweredPriority1 && !loweredPriority2) {
+            ret = 1;
+        } else if (!loweredPriority1 && loweredPriority2) {
+            ret = 0;
+        } else if (ability1 == ABILITY_STALL && ability2 == ABILITY_STALL) {
+            if (speed1 > speed2) {
+                ret = 1;
+            } else if (speed1 == speed2 && BattleSys_Random(bsys)&1) {
+                ret = 2;
+            }
+        } else if (ability1 == ABILITY_STALL && ability2 != ABILITY_STALL) {
+            ret = 1;
+        } else if (ability1 != ABILITY_STALL && ability2 == ABILITY_STALL) {
+            ret = 0;
+        } else if (ctx->fieldCondition & FIELD_CONDITION_TRICK_ROOM) {
+            if (speed1 > speed2) {
+                ret = 1;
+            }
+            if (speed1 == speed2 && BattleSys_Random(bsys)&1) {
+                ret = 2;
+            }
+        } else {
+            if (speed1 < speed2) {
+                ret = 1;
+            }
+            if (speed1 == speed2 && BattleSys_Random(bsys)&1) {
+                ret = 2;
+            }
+        }
+    } else if (movePriority1 < movePriority2) {
+        ret = 1;
+    }
+    
+    return ret;
+}
```

Comment:
newline at the end of the file

Reply:
this file still has ~17k more lines of asm to get through so it'll be there lmao